proto/spacemesh/api.proto

```diff
@@ -1,166 +1,70 @@
-}
+    // Tell the node to start syncing
+    rpc NodeSyncStart(google.protobuf.Empty) returns (google.protobuf.Empty);
 
-message LayerId {
-    bytes id = 1;
-}
+    // Gracefully shut down the node
+    rpc NodeShutdown(google.protobuf.Empty) returns (google.protobuf.Empty);
 
-message Layer {
-    LayerId id = 1;
-    repeated Block blocks = 2;
-}
+    ///////// Mesh facade
 
-message LayerEvent {
-    LayerId layer_id = 1;
-    enum Event {
-        UNDEFINED = 0;
-        APPROVED = 1;
-        CONFIRMED = 2;
-    }
-}
+    // Returns mesh genesis time as unix epoch time
+    rpc MeshGenesisTime(google.protobuf.Empty) returns (SimpleInt);
 
-message AccountEvent {
-    enum Event {
-        UNDEFINED = 0;
-        BALANCE_CHANGE = 1;
-        NONCE_CHANGE = 2;
-    }
-    uint64 value = 1;
-}
+    // Returns current layer number
+    rpc MeshCurrentLayer(google.protobuf.Empty) returns (SimpleInt);
 
-message NodeStatus {
-    uint64 peers = 1;
-    uint64 min_peers = 2;
-    uint64 max_peers = 3;
-    bool synced = 4;
-    uint64 synced_layer = 5;
-    uint64 current_layer = 6;
-    uint64 verified_layer = 7;
-}
+    // Returns current epoch number
+    rpc MeshCurrentEpoch(google.protobuf.Empty) returns (SimpleInt);
 
-/*
-   Represents a critical node error.
-   TODO: add an enum with clearly-defined error types.
- */
-message NodeError {
-    string error = 1;
-}
+    // Returns network ID
+    rpc MeshNetId(google.protobuf.Empty) returns (SimpleInt);
 
-message SyncStatus {
-  enum SyncStatus {
-      UNDEFINED = 0;
-      NEW_LAYER_VERIFIED = 1;
-      NEW_TOP_LAYER = 2;
-      SYNCED = 3;
-      SYNCING = 4;
-      NOT_SYNCED = 5;
-  }
-}
+    // Returns number of layers per epoch (a network constant)
+    rpc MeshEpochNumLayers(google.protobuf.Empty) returns (SimpleInt);
 
-service SpacemeshService {
-    // SCALAR
+    ////////// Node streaming data facade
 
-    // Test facade
+    // Sync status events
+    rpc NodeSyncStatuses(google.protobuf.Empty) returns (stream NodeSyncStatus);
 
-    // A simple test endpoint
-    rpc Echo(SimpleMessage) returns (SimpleMessage);
+    // Node error events
+    rpc NodeErrors(google.protobuf.Empty) returns (stream NodeError);
 
-    // Node facade
+    //////////////// Mesh streaming data facade
 
-    // Returns the version of the node software as a semver string
-    rpc NodeVersion(google.protobuf.Empty) returns (SimpleMessage);
-    // Returns node status object
-    rpc NodeStatus(google.protobuf.Empty) returns (NodeStatus);
-    // Tell the node to start sync
-    rpc NodeSyncStart(google.protobuf.Empty) returns (google.protobuf.Empty);
+    // Receipts - emitted after tx was processed by STF (or rejected before STF)
+    rpc MeshTransactionReceipts(google.protobuf.Empty) returns (stream TransactionReceipt);
 
-    // Mesh facade
+    // App Events - emitted by app methods impl code trigged by an app transaction
+    rpc MeshAppEvents(google.protobuf.Empty) returns (stream AppEvent);
```

Comment:
Think we should leave this out from the current API version - this is part of the VM - can we bump to 0.0.2?

Reply:
We are going to have smart wallets in 0.2 and events and receipts are quite isolated but we can deal with these in 0.0.2 as you describe as long as we will provide them for this project - the explorer version spec includes full apps events, receipts and state...